### PR TITLE
Created Students Show View

### PR DESCRIPTION
**Pull request by MatiasMercado**
## Students Show View

The view shows a student's detail. It can be access at /students/:docket.
Since student is hard coded, the view can be accessed using any docket.

## Trello Card
https://trello.com/c/c16h7OXj/34-maquetar-vista-de-un-alumno

### CODE REVIEW

File: app/scripts/controllers/StudentsShowCtrl.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+define(['paw'], function(paw) {
+  paw.controller('StudentsShowCtrl', ['$routeParams', function($routeParams) {
```
Comment (MatiasComercio) [inline app/scripts/controllers/StudentsShowCtrl.js]: Fix indentation

Comment (MatiasComercio) [inline app/scripts/controllers/StudentsShowCtrl.js]: Nice use of the array and string service injection! :smile:

File: app/scripts/controllers/StudentsShowCtrl.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    var _this = this;
+
+    this.docket = $routeParams.docket; // For future Service calls
```
Comment (MatiasComercio) [inline app/scripts/controllers/StudentsShowCtrl.js]: Perfect! :+1: 

However, I wouldn't assign this to the controller, but to a variable, as it is not necessary to expose this information to the view.
i.e.: 

```js
var docket = $routeParams.docket; // For future Service calls
```

File: app/specs/controllers/StudentsShowCtrl.spec.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    beforeEach(module('paw'));
+
+    var $controller, controller, expectedStudent;
+
+    beforeEach(inject(
+      function(_$controller_) {
+        $controller = _$controller_;
+        controller = $controller('StudentsShowCtrl');
+        expectedStudent =
+        {
```
Comment (MatiasComercio) [inline app/specs/controllers/StudentsShowCtrl.spec.js]: Unify style: either you put this `{` beside the  above `=` or you fix the `address: {` line

File: app/specs/controllers/StudentsShowCtrl.spec.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            door: 'A',
+            telephone: '1544683390',
+            zipCode: '1100'
+          }
+        };
+      }));
+
+      it('loads the required student', function() {
+        expect(controller.student).toEqual(expectedStudent);
+      });
```
Comment (MatiasComercio) [inline app/specs/controllers/StudentsShowCtrl.spec.js]: Test case missing: you should be testing that the controller's docket is being loaded from the $routeParams service. Take a look at the `StudentsIndexCtrl.spec.js` from #63 

However, if you've changed the docket assignment from `this.docket` to `var docket` as suggested above, you won't be able to test the controllers variable (as it is local to that function). What you would have to do is to use a [Jasmine Spy](https://jasmine.github.io/edge/introduction.html#section-Spies) and check that the $routeParam.docket is being requested. You can also check other spec files for further help. If you cannot manage it, just tell me.

File: app/views/students/show.html [hunk 5]
```diff
@@ -0,0 +1,125 @@
+<div ng-controller='StudentsShowCtrl as controller'>
+	<!-- Page Heading -->
+	<div class="row">
+		<div class="col-xs-12">
+			<h2 class="page-header">
```
Comment (MatiasComercio) [inline app/views/students/show.html]: I know that I've told you to use `h2` as a page header in #63, but I made a mistake. Sorry for that. Please update this to `h1`

File: app/views/students/show.html [hunk 6]
```diff
@@ -0,0 +1,125 @@
+<div ng-controller='StudentsShowCtrl as controller'>
```
Comment (MatiasComercio) [inline app/views/students/show.html]: Use simple quotes whenever possible. Ask me for private how to do it quickly.

File: app/views/students/show.html [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+	<!-- Page Heading -->
+	<div class="row">
+		<div class="col-xs-12">
+			<h2 class="page-header">
+        <span translate="i18nStudentsPanelSection"></span>
```
Comment (MatiasComercio) [inline app/views/students/show.html]: Try to create another translation with the same text. I know, maybe it sounds kind of weird, but let me explain you.

Your string representing your translation should represent where you will use it, no matter what its text is. So, in this case, I would define a translation
` i18nStudents: Alumnos`. 

Trade off of this option: cleaner and semantic code, and independent key-value translation per view vs. wording repetition.

File: app/scripts/routes.js [hunk 8]
```diff
@@ -11,6 +11,10 @@ define([], function() {
             '/login': {
                 templateUrl: '/views/login.html',
                 controller: 'LoginCtrl'
+            },
+            '/student/:docket': {
```
Comment (MatiasComercio) [inline app/scripts/routes.js]: This path is not OK. it should be `/students/:docket` ('s' from students missing)

File: app/views/students/show.html (hunk 7 above)
Comment (MatiasMercado) [inline app/views/students/show.html]: I'm not even sure why it's there. It should the the student's firstName + lastName.

File: app/specs/controllers/StudentsShowCtrl.spec.js [hunk 9]
```diff
@@ -12,36 +12,39 @@ function() {
   describe('Students Show Ctrl', function() {
     beforeEach(module('paw'));
 
-    var $controller, controller, expectedStudent;
+    var expectedStudent = expectedStudent = {
```
Comment (MatiasComercio) [inline app/specs/controllers/StudentsShowCtrl.spec.js]: What's this? I haven't seen it before...

I'll change it, but if you intentionally wrote this, please explain it to me.

Comment (MatiasMercado) [inline app/specs/controllers/StudentsShowCtrl.spec.js]: Sorry, definitely not intentional.

Comment (MatiasComercio) [inline app/specs/controllers/StudentsShowCtrl.spec.js]: No problem bro! :smile:

**Pull request merged**